activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -67,7 +67,7 @@ def sql_color(sql)
       case sql
         when /\A\s*rollback/mi
           RED
-        when /\s*.*?select .*for update/mi, /\A\s*lock/mi
+        when /select .*for update/mi, /\A\s*lock/mi
```

Comment:
Unless I'm missing something, the original code will also match @fillman 's example. As it is, with normal regexps, I believe it's impossible to always be 100% correct.
```ruby
bash-3.2$ irb
2.3.0 :001 > s = "INSERT ... VALUES (
SELECT * FROM users WHERE users.name = 'for update'
)
..."
2.3.0 :002"> 2.3.0 :003"> 2.3.0 :004">  => "INSERT ... VALUES (\nSELECT * FROM users WHERE users.name = 'for update'\n)\n..." 
2.3.0 :005 > s
 => "INSERT ... VALUES (\nSELECT * FROM users WHERE users.name = 'for update'\n)\n..." 
2.3.0 :006 > puts s
INSERT ... VALUES (
SELECT * FROM users WHERE users.name = 'for update'
)
...
 => nil 
2.3.0 :007 > s.match(/\s*.*?select .*for update/mi)
 => #<MatchData "INSERT ... VALUES (\nSELECT * FROM users WHERE users.name = 'for update"> 
```